node/cn/snap/handler.go

```diff
@@ -0,0 +1,524 @@
+	if req.Bytes > softResponseLimit {
+		req.Bytes = softResponseLimit
+	}
+	// TODO-Klaytn-SnapSync Do we want to enforce > 0 accounts and 1 account if origin is set?
+	// TODO-Klaytn-SnapSync   - Logging locally is not ideal as remote faulst annoy the local user
+	// TODO-Klaytn-SnapSync   - Dropping the remote peer is less flexible wrt client bugs (slow is better than non-functional)
+
+	// Calculate the hard limit at which to abort, even if mid storage trie
+	hardLimit := uint64(float64(req.Bytes) * (1 + stateLookupSlack))
+
+	// Retrieve storage ranges until the packet limit is reached
+	var (
+		slots  = make([][]*StorageData, 0, len(req.Accounts))
+		proofs [][]byte
+		size   uint64
+	)
+	for _, accountHash := range req.Accounts {
+		// If we've exceeded the requested data limit, abort without opening
+		// a new storage range (that we'd need to prove due to exceeded size)
+		if size >= req.Bytes {
+			break
+		}
+		// The first account might start from a different origin and end sooner
+		var origin common.Hash
+		if len(req.Origin) > 0 {
+			origin, req.Origin = common.BytesToHash(req.Origin), nil
+		}
+		limit := common.HexToHash("0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff")
+		if len(req.Limit) > 0 {
+			limit, req.Limit = common.BytesToHash(req.Limit), nil
+		}
+		// Retrieve the requested state and bail out if non existent
+		it, err := chain.Snapshots().StorageIterator(req.Root, accountHash, origin)
+		if err != nil {
+			return nil, nil
+		}
+		// Iterate over the requested range and pile slots up
+		var (
+			storage []*StorageData
+			last    common.Hash
+			abort   bool
+		)
+		for it.Next() {
+			if size >= hardLimit {
+				abort = true
+				break
+			}
+			hash, slot := it.Hash(), common.CopyBytes(it.Slot())
+
+			// Track the returned interval for the Merkle proofs
+			last = hash
+
+			// Assemble the reply item
+			size += uint64(common.HashLength + len(slot))
+			storage = append(storage, &StorageData{
+				Hash: hash,
+				Body: slot,
+			})
+			// If we've exceeded the request threshold, abort
+			if bytes.Compare(hash[:], limit[:]) >= 0 {
+				break
+			}
+		}
+		slots = append(slots, storage)
+		it.Release()
+
+		// Generate the Merkle proofs for the first and last storage slot, but
+		// only if the response was capped. If the entire storage trie included
+		// in the response, no need for any proofs.
+		if origin != (common.Hash{}) || abort {
+			// Request started at a non-zero hash or was capped prematurely, add
+			// the endpoint Merkle proofs
+			accTrie, err := statedb.NewTrie(req.Root, chain.StateCache().TrieDB())
+			if err != nil {
+				return nil, nil
+			}
+			serializer := account.NewAccountSerializer()
+			if err := rlp.DecodeBytes(accTrie.Get(accountHash[:]), serializer); err != nil {
+				return nil, nil
+			}
+			acc := serializer.GetAccount()
+			pacc := account.GetProgramAccount(acc)
+			if pacc == nil {
+				return nil, nil
```

Comment:
Do we need to return? I think we can skip this account if it is not a program account.

Reply:
As we talked let's keep this for now, and it can be enhanced later if necessary.